### PR TITLE
:art: Pass abstract filesystem to sysl.Parse() + more testable main()

### PR DESCRIPTION
**Pull request by anzdaddy**
Changes proposed in this pull request:
- Replace direct os calls in `sysl.Parse` with `http.FileSystem` parameter.
- Split main into three functions to improve support for testing.

@anz-bank/sysl-developers

### CODE REVIEW

**Comment by anzdaddy** [comment]
Removed reviewers while fixing CI breaks.

**Comment by anzdaddy** [comment]
@juliaogris Any idea why codecov is happy with #166 but not this PR? They both report 70.18% coverage. (Maybe it'll go away once I merge #166.)

**Comment by anzdaddy** [comment]
For some reason all the new ANTLR-generated JavaScript code was being picked up by codecov in this PR but not in #166. I've added a `.codecov.yml` to ignore generated JS.

File: sysl2/sysl/grammar/lexer_impl.go [hunk 1]
```diff
@@ -102,6 +104,9 @@ func GetNextToken(l *SyslLexer) antlr.Token {
 	}
 
 	next := l.BaseLexer.NextToken()
+	if os.Getenv("SYSL_LEXER_LOG") != "" {
```
Comment (sahejsingh) [inline sysl2/sysl/grammar/lexer_impl.go]: Shouldn't check env var for each token!

**Pull request merged**